extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -104,18 +111,21 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm) thro
     protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, RunnableWithException afterEndTransaction)
             throws Exception {
 
-        TransactionConfiguration configAnnotation = getTransactionConfiguration(ic);
+        Integer timeoutConfiguredForMethod = cache.computeIfAbsent(ic.getMethod(),
+                (m) -> extractTransactionConfigurationTimeoutFromAnnotation(ic));
```

Comment:
We try to avoid lambdas in runtime code as they consume memory.

Also given it's a sensitive call site, better do a `get()` and return it if not null before doing the `computeIfAbsent()` (this would need to be moved to a method).

Now, I'm not exactly sure what cost this will have even when fully optimized.

Reply:
@gsmet `computeIfAbsent` is right here, if you do a get first then compute the missing value you exposes yourself on concurrency issues. 

The lambda can be an issue here as it's a capturing lambda and those have memory consideration, but I'm not against them (I know people on the community don't like them) it has a very slight impact with respect with transaction management ;)

Reply:
I'm not saying `computeIfAbsent` shouldn't be used. I'm saying you should do a get before and return directly if not null. You do the `computeIfAbsent` after that. It's a CHM, you won't have concurrency issues doing that. Locking is less efficient when using computeIfAbsent so if you can avoid calling it, it's a win. And here, at some point, all the values will be in the cache so, after the warmup, things will be significantly better because you won't call `computeIfAbsent()` at all. @Sanne was the one mentioning this to me first and I did numerous HV benchmarks proving this point (at least with the JDK used at that time).

As for lambdas, we decided as a community to avoid them as much as possible in runtime code (i.e. except if they really bring something to the plate. This is not the case here.

Reply:
After giving it some additional thoughts, I think the cache should be only used for methods annotated with `timeoutFromConfigProperty`. It won't cost that much to not use the cache in other cases and it will avoid the cost of it entirely. My wild guess is that having the cache will have a higher cost if either we don't have anything set or if we are just using `timeout`.

@marcelhanser and you thought it would be easy ;)

Reply:
I thought i do a small little pull-request and go to get a beer - buuuut^^
hmm... How do we know if a method is annotated with timeoutFromConfigProperty, without looking into it and do the costly reflection call? That was the reason for me to cache 'all'

Reply:
@gsmet oh yeah, I remember already having this discussion, sorry not to have remembered it ;)

So yes, making a get() before is good, there is a slight risk to compute two times the cache entry but it should not be an issue here as computing it is not very ressource consuming.

> After giving it some additional thoughts, I think the cache should be only used for methods annotated with timeoutFromConfigProperty. 

Retrieving annotation is costly I think, it's reflection so it may be cached, but still ... I remember we discuss it with @Sanne and agree that at some point we will cache the `TransactionConfiguration` to avoid too much reflection.
So the correct thing to cache is not the timeout but the full `TransactionConfiguration` object to avoid the reflective all.

Reply:
Hum, forget what I just told, caching the annotation will not be usefull here as we want to avoid the call to the config ...

So I'll keep it as is, or create a `TransactionConfigurationInfo` object that will have the informations resolved from the annotation, as the annotation currently only deals with timeout it may not be necessary.

Reply:
> So yes, making a get() before is good, there is a slight risk to compute two times the cache entry but it should not be an issue here as computing it is not very ressource consuming.

There isn't a race: first you try `get()`, when it fails you still do the `computeIfAbsent` so you'll have a lock. Worst thing that can happen is that the initial call does an additional pointless `get()` ... totally ok.